nncf/experimental/onnx/graph/onnx_graph.py

```diff
@@ -202,24 +206,102 @@ def get_weight_tensor(self, node: onnx.NodeProto) -> Tuple[str, np.ndarray]:
         :param node: Node, in which the weight tensor finally applied.
         :return: Weight tensor name and its value.
         """
-        node_name = node.name
-        node_inputs = self.get_node_edge_names(node_name)['input']
-        weight_input = node_inputs[1]
-        try:
+        weight_port_id = self.get_weight_port_id(node)
+        weight_input = self.get_node_edge_names(node.name)['input'][weight_port_id]
+        if self.has_initializer(weight_input):
             return self.get_initializer(weight_input).name, self.get_initializers_value(weight_input)
-        except RuntimeError as e:
-            node = self.get_nodes_by_output(weight_input)[0]
-            metatype = ONNX_OPERATION_METATYPES.get_operator_metatype_by_op_name(node.op_type)
-            if metatype == ONNXIdentityMetatype:
-                tensor_name = self.get_initializer(node.input[0]).name
-                return tensor_name, self.get_initializers_value(tensor_name)
+        weight_port_id = self.get_weight_port_id(node)
+        parent_node_on_weight_port = self.get_parents(node)[weight_port_id]
+        nodes = deque([parent_node_on_weight_port])
+        while nodes:
+            current_node = nodes.popleft()
+            node_parents = self.get_parents(current_node)
+            nodes.extendleft(node_parents)
+            metatype = ONNX_OPERATION_METATYPES.get_operator_metatype_by_op_name(current_node.op_type)
+            if metatype in [ONNXIdentityMetatype, ONNXQuantizeLinearMetatype]:
+                if self.has_initializer(current_node.input[0]):
+                    return self._get_tensor_from_zero_input(current_node)
+                continue
```

Comment:
This implementation is not general. It works only for a limited scope of models. It looks like a hack.

Reply:
Yes, it is. This functional will be extended and improved in the following PRs. I placed TODO

Reply:
Alright. But this code will not be moved from the experimental until this will not have been fixed.